Tests/Rename-CIisAppPool.Tests.ps1

```diff
@@ -0,0 +1,99 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    Set-StrictMode -Version 'Latest'
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    function GivenAppPool
+    {
+        param(
+            [Parameter(Mandatory, Position=0)]
+            [String] $Named
+        )
+
+        Install-CIisAppPool -Name $Named
+    }
+
+    function ThenAppPool
+    {
+        param(
+            [Parameter(Mandatory, Position=0)]
+            [String] $Named,
+
+            [switch] $Not,
+
+            [Parameter(Mandatory)]
+            [switch] $Exists
+        )
+
+        $mgr = [Microsoft.Web.Administration.ServerManager]::New()
+        $appPool = $mgr.ApplicationPools | Where-Object 'name' -EQ $Named
+        if ($Not -and $Exists)
+        {
+            $appPool | Should -BeNullOrEmpty
+        }
+        elseif ($Exists)
+        {
+            $appPool | Should -Not -BeNullOrEmpty

```

Comment:
I think you can eliminate the conditional by just doing:

```powershell
$appPool | Should -Not:$Not -BeNullOrEmpty
```

Reply:
Agreed, but this is a conscience style choice for me: I like my tests and assertions to read like a sentence and having optional/conditional parameters breaks that flow.